File: src/main/java/com/cardlay/techinterview/cardmanagement/controller/request/CreateCardDto.java
package com.cardlay.techinterview.cardmanagement.controller.request;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

public record CreateCardDto(
    @NotNull
    @NotBlank
    String cardNumber,

    @Min(0)
    BigDecimal balance
) {
}

File: src/main/kotlin/com/cardlay/techinterview/cardmanagement/controller/request/CreateCardDto.kt
package com.cardlay.techinterview.cardmanagement.controller.request

import java.math.BigDecimal
import javax.validation.constraints.Min
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class CreateCardDto(
    @field:NotNull
    @field:NotBlank
    val cardNumber: String,

    @field:Min(0)
    val balance: BigDecimal,
)